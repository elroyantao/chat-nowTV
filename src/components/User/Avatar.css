.Avatar {
  width: 100%;
  height: 100%;
  border-radius: 50%;
  overflow: hidden;
  display: flex;
  justify-content: center;
  align-items: center;
}
.Avatar-image {
  text-align: center;
}