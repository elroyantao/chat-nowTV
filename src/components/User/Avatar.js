import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './Avatar.css';

export default class Avatar extends Component {
  static propTypes = {
    avatar: PropTypes.string,
    alt: PropTypes.string
  }
  defaultProps = {
    alt: ''
  }
  render() {
    const { avatar, alt } = this.props
    if (!avatar && !alt) return null
    return (
      <div className="Avatar">
        <img alt={alt} src={avatar} className="Avatar-image"/>
      </div>
    )
  }
}