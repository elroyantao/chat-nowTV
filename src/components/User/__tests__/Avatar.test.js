import testHelper from '../../../helpers/test-helper'
import Avatar from '../Avatar'

describe('<Avatar/>', () => {
  const renderComponent = testHelper(Avatar)
  const props = {
    avatar: 'http://dummyimage.com/100x100.bmp/cc0000/ffffff',
    alt: 'Tom'
  }
  describe('@renders', () => {
    it('in default state', () => {
      expect(renderComponent(props).getTree()).toMatchSnapshot()
    })
    it('when no avatar', () => {
      expect(renderComponent({
        ...props,
        avatar: undefined
      }).getTree()).toMatchSnapshot()
    })
    it('when no avatar and alt', () => {
      expect(renderComponent({
        ...props,
        avatar: undefined,
        alt: undefined
      }).getTree()).toMatchSnapshot()
    })
  })
})