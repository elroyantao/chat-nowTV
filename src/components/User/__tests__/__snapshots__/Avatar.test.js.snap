exports[`<Avatar/> @renders in default state 1`] = `
<div
  className="Avatar">
  <img
    alt="Tom"
    className="Avatar-image"
    src="http://dummyimage.com/100x100.bmp/cc0000/ffffff" />
</div>
`;

exports[`<Avatar/> @renders when no avatar 1`] = `
<div
  className="Avatar">
  <img
    alt="Tom"
    className="Avatar-image" />
</div>
`;

exports[`<Avatar/> @renders when no avatar and alt 1`] = `""`;
